quantlib/instruments/credit_default_swap.pyx

```diff
@@ -178,7 +179,13 @@ cdef class CreditDefaultSwap(Instrument):
 
     @property
     def coupons(self):
-        return cashflow.leg_items(_get_cds(self).coupons())
+        cdef vector[shared_ptr[_cf.CashFlow]] coupons = _get_cds(self).coupons()
+        cdef vector[shared_ptr[_cf.CashFlow]].iterator it = coupons.begin()
+        cdef _cf.CashFlow* _thiscf
+        while it != coupons.end():
+            _thiscf = deref(it).get()
+            yield (_thiscf.amount(), _pydate_from_qldate(_thiscf.date()))
```

Comment:
I wonder if having a generator here is not going to be counter intuitive for the users. Any  specific reasons why you chose this option rather than return a list?